int64 IR_1
int64 IR_2
int64 IR_3
int64 IR_4
int64 IR_5
---
string direction